### ISSUE TITLE
Config file exists in project root directory but seems ignored

### ISSUE THREAD

**Issue by mloskot** [terraform-linters/tflint#1705]
### Summary

The `tflint --help` says `Config file name (default: .tflint.hcl)` which I take as meaning that `.tflint.hcl` is looked up for and read automatically by default.

There is `.tflint.hcl` exists in top-level folder of my Terraform project, but `tflint` seems to be ignoring it unless I specify it explicitly.

### Command

tflint

### Terraform Configuration

```terraform
N/A
```

### TFLint Configuration

```HCL
`cat .tflint.hcl` says


plugin "terraform" {
  enabled = true
  preset  = "recommended"
}

plugin "azurerm" {
    enabled = true
    version = "0.21.0"
    source  = "github.com/terraform-linters/tflint-ruleset-azurerm"
}

rule "terraform_typed_variables" {
  enabled = false
}
```

### Output

```Shell
Run with implicit/automatic `.tflint.hcl` loading - expected output `terraform_typed_variables` warnings filtered out


$ tflint --format compact --recursive | grep terraform_typed_variables | wc -l
87


Run with explicit `.tflint.hcl` via filename only - expected output `terraform_typed_variables` warnings filtered out

```console
$ tflint --format compact --recursive --config .tflint.hcl | grep terraform_typed_variables | wc -l
87
```

Run with explicit `.tflint.hcl` via absolute path - Success! All `terraform_typed_variables` warnings filtered

```console
$ tflint --format compact --recursive --config $PWD/.tflint.hcl | grep terraform_typed_variables | wc -l
0
```

Interestingly, if I give the config file but via relative path, then I get this error:

```console
$ tflint --format compact --recursive --config ./.tflint.hcl
Failed to load TFLint config; failed to load file: open ./.tflint.hcl: no such file or directory
```
```


### TFLint Version

0.45.0

### Terraform Version

1.3.7

### Operating System

- [X] Linux
- [ ] macOS
- [X] Windows
